app/models/miq_ae_namespace.rb

```diff
@@ -11,27 +11,38 @@ class MiqAeNamespace < ApplicationRecord
                          /^commit_time/, /^commit_sha/, /^ref$/, /^ref_type$/,
                          /^last_import_on/, /^source/, /^top_level_namespace/].freeze
 
+  belongs_to :domain, :class_name => "MiqAeDomain", :inverse_of => false
   has_many :ae_classes, -> { includes([:ae_methods, :ae_fields, :ae_instances]) }, :class_name => "MiqAeClass",
-           :foreign_key => :namespace_id, :dependent => :destroy, :inverse_of => false
+           :foreign_key => :namespace_id, :dependent => :destroy, :inverse_of => :ae_namespace
 
   validates :name,
             :format     => {:with => /\A[\w\.\-\$]+\z/i, :message => N_("may contain only alphanumeric and _ . - $ characters")},
             :presence   => true,
             :uniqueness => {:scope => :ancestry, :case_sensitive => false}
 
-  alias ae_namespaces children
+  alias_attribute :fqname_sans_domain, :relative_path
   virtual_has_many :ae_namespaces
+  alias ae_namespaces children
+
+  before_validation :set_relative_path
+  after_save :set_children_relative_path
+
+  def parent
+    parent_id == domain_id ? domain : super
+  end
 
   def self.lookup_by_fqname(fqname, include_classes = true)
     return nil if fqname.blank?
 
-    fqname = fqname[0] == '/' ? fqname : "/#{fqname}"
-    fqname = fqname.downcase
-    last = fqname.split('/').last
-    low_name = arel_table[:name].lower
+    fqname = fqname[1..-1] if fqname[0] == '/'
+    dname, *partial = fqname.downcase.split('/')
```

Comment:
Super-minor but it feels like these 2 line should be pulled into a helper method, like `domain_name_from_fqname(fqname)` or something.